demo/src/main/scala/dev/atedeg/ecscalademo/systems/MovementSystem.scala

```diff
@@ -0,0 +1,24 @@
+package dev.atedeg.ecscalademo.systems
+
+import dev.atedeg.ecscala.*
+import dev.atedeg.ecscala.util.types.given
+import dev.atedeg.ecscala.CNil
+import dev.atedeg.ecscalademo.{ PlayState, Position, Velocity }
+import dev.atedeg.ecscalademo.*
+
+/**
+ * The [[System]] that updates the balls Positions given the updated Velocities
+ */
+class MovementSystem extends System[Position &: Velocity &: CNil] {
+  override def shouldRun: Boolean = PlayState.playing
+
+  override def update(entity: Entity, components: Position &: Velocity &: CNil)(
+      deltaTime: DeltaTime,
+      world: World,
+      view: View[Position &: Velocity &: CNil],
+  ): Deletable[Position &: Velocity &: CNil] = {
+    val Position(point) &: Velocity(vector) &: CNil = components
```

Comment:
To me it read better like this
```suggestion
    val Position(position) &: Velocity(velocity) &: CNil = components
```